pkg/webui/console/views/device-overview/index.js

```diff
@@ -219,21 +242,30 @@ class DeviceOverview extends React.Component {
   }
 
   render() {
-    const { device } = this.props
+    const { device, shouldRedirect } = this.props
     const devIds = device && device.ids
+    const otherwise = {
+      redirect: '/applications',
+      render: toast({
+        type: toast.types.WARNING,
+        message: m.failedAccessOtherHostDeviceMessage,
```

Comment:
This won't work as expected. The way that the `<Redirect />`\-component currently works is either `redirect` or `render`. Specifying both will give precedence to the redirect. Please make sure to check implementation details or test manually before using modules. I think it makes sense to add a `message` prop to the `otherwise` object to launch a toast message in any case if the requirement is not met.

```javascript
const otherwise = {
  redirect: '/applications',
  message: m.failedAccessOtherHostDeviceMessage,
}
```

You'll probably also want to update the respective cypress test to check for the toast notification.

Reply:
That's weird. I tested it manually and a toast appeared right after the redirect. It doesn't work for you? What could be the reason for the precedence to not be working correctly?

Reply:
It is indeed working because as part of defining the `otherwise` prop, you are invoking the `toast()` function, which does not return a compnent but in itself will triggers a toast. This also means that the toast will appear for any render of regular end device overviews that do not redirect.

Reply:
It makes sense. Thanks for explaining. I will add a `message` prop to the `otherwise` object.